### PR TITLE
[proxima-core] make RenameTransformation public

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

File: core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import cz.o2.proxima.annotations.Evolving;
+import cz.o2.proxima.functional.BiFunction;
+import cz.o2.proxima.functional.UnaryFunction;
+import cz.o2.proxima.repository.AttributeDescriptor;
+import cz.o2.proxima.repository.Repository;
+import cz.o2.proxima.storage.StreamElement;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+
+/** A stateless transformation that performs transformation from one attribute to another. */
```
Comment (LesTR) [inline core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java]: Can be this javadoc more detailed please? Ideally with usage example?

File: core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  @Override
+  public int apply(StreamElement input, Collector<StreamElement> collector) {
+    @SuppressWarnings("unchecked")
+    AttributeDescriptor<Object> desc =
+        (AttributeDescriptor<Object>) descTransform.apply(input.getAttributeDescriptor());
+
+    if (input.isDelete()) {
+      collector.collect(
+          input.isDeleteWildcard()
+              ? StreamElement.deleteWildcard(
```
Comment (LesTR) [inline core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java]: I'll prefer here normal `if` statement which will be imho more readable.

File: core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          StreamElement.upsert(
+              input.getEntityDescriptor(),
+              desc,
+              input.getUuid(),
+              input.getKey(),
+              nameTransform.apply(input.getAttribute(), input.getAttributeDescriptor()),
+              input.getStamp(),
+              remapPayload(input, desc)));
+    }
+    return 1;
```
Comment (LesTR) [inline core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java]: Notice: this return code is still not clear for me :(

Comment (je-ik) [reply core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java]: This is due to the fact, that the rename can be asynchronous, and therefore the collector might not get invoked prior to returning from this method. The return value signals how many elements to wait for.

File: core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java (hunk 2 above)
Comment (je-ik) [reply core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java]: :+1:

File: core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java (hunk 1 above)
Comment (je-ik) [reply core/src/main/java/cz/o2/proxima/transform/RenameTransformation.java]: Done.

**Pull request merged**